ui/app/components/auth-jwt.js

```diff
@@ -0,0 +1,165 @@
+import Ember from 'ember';
+import { inject as service } from '@ember/service';
+import Component from './outer-html';
+import { next, later } from '@ember/runloop';
+import { task, timeout, waitForEvent } from 'ember-concurrency';
+import { computed } from '@ember/object';
+
+const WAIT_TIME = Ember.testing ? 0 : 500;
+const ERROR_WINDOW_CLOSED =
+  'The provider window was closed before authentication was complete.  Please click Sign In to try again.';
+const ERROR_MISSING_PARAMS =
+  'The callback from the provider did not supply all of the required parameters.  Please click Sign In to try again. If the problem persists, you may want to contact your administrator.';
+
+export { ERROR_WINDOW_CLOSED, ERROR_MISSING_PARAMS };
+
+export default Component.extend({
+  store: service(),
+  selectedAuthPath: null,
+  roleName: null,
+  role: null,
+  onRoleName() {},
+  onLoading() {},
+  onError() {},
+  onToken() {},
+  onNamespace() {},
+
+  didReceiveAttrs() {
+    next(() => {
+      let { oldSelectedAuthPath, selectedAuthPath } = this;
+      if (oldSelectedAuthPath !== selectedAuthPath) {
+        this.set('role', null);
+        this.onRoleName(null);
+        this.fetchRole.perform(null, { debounce: false });
+      }
+      this.set('oldSelectedAuthPath', selectedAuthPath);
+    });
+  },
+
+  // OIDC roles in the JWT/OIDC backend are those with an authUrl,
+  // those that are JWT type will 400 when trying to fetch the role
+  isOIDC: computed('role', 'role.authUrl', function() {
+    return this.role && this.role.authUrl;
+  }),
+
+  getWindow() {
+    return this.window || window;
+  },
+
+  fetchRole: task(function*(roleName, options = { debounce: true }) {
+    if (options.debounce) {
+      this.onRoleName(roleName);
+      // debounce
+      yield timeout(WAIT_TIME);
+    }
+    let path = this.selectedAuthPath || 'jwt';
+    let id = JSON.stringify([path, roleName]);
+    let role = null;
+    try {
+      role = yield this.store.findRecord('role-jwt', id, { adapterOptions: { namespace: this.namespace } });
+    } catch (e) {
+      if (!e.httpStatus || e.httpStatus !== 400) {
+        throw e;
+      }
+    }
+    this.set('role', role);
+  }).restartable(),
+
+  handleOIDCError(err) {
+    this.onLoading(false);
+    this.prepareForOIDC.cancelAll();
+    this.onError(err);
+  },
+
+  prepareForOIDC: task(function*(oidcWindow) {
```

Comment:
ah, the use of ember concurrency tasks here is really slick and easy to follow. nice! 💅